src/main/webapp/app/entities/programming-exercise/actions/programmming-exercise-instructor-submission-state.component.ts

```diff
@@ -50,23 +58,64 @@ export class ProgrammmingExerciseInstructorSubmissionStateComponent implements O
     }
 
     triggerBuildOfAllSubmissions() {
-        this.programmingSubmissionService.triggerInstructorBuildForAllParticipationsOfExercise(this.exerciseId).subscribe();
+        const modalRef = this.modalService.open(ProgrammingExerciseInstructorTriggerAllDialogComponent, { size: 'lg', backdrop: 'static' });
+        modalRef.componentInstance.exerciseId = this.exerciseId;
     }
 
     /**
      * Retrieve the participation ids that have a failed submission and retry their build.
      */
     triggerBuildOfFailedSubmissions() {
         this.isBuildingFailedSubmissions = true;
-        const failedSubmissionParticipations = this.programmingSubmissionService.getFailedSubmissionParticipationsForExercise(this.exerciseId);
+        const failedSubmissionParticipations = this.programmingSubmissionService.getSubmissionCountByType(this.exerciseId, ProgrammingSubmissionState.HAS_FAILED_SUBMISSION);
         this.programmingSubmissionService
             .triggerInstructorBuildForParticipationsOfExercise(this.exerciseId, failedSubmissionParticipations)
-            .pipe()
             .subscribe(() => (this.isBuildingFailedSubmissions = false));
     }
 
-    private sumSubmissionStates = (buildState: ExerciseBuildState) =>
-        Object.values(buildState).reduce((acc: { [state: string]: number }, [submissionState]) => {
+    private sumSubmissionStates = (buildState: ExerciseSubmissionState) =>
+        Object.values(buildState).reduce((acc: { [state: string]: number }, { submissionState }) => {
             return { ...acc, [submissionState]: (acc[submissionState] || 0) + 1 };
         }, {});
 }
+
+@Component({
+    template: `
+        <form name="triggerAllForm" (ngSubmit)="confirmTrigger()">
+            <div class="modal-header">
+                <h4 class="modal-title" jhiTranslate="artemisApp.programmingExercise.resubmitAll">Trigger all</h4>
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true" (click)="cancel()">&times;</button>
+            </div>
+            <div class="modal-body">
+                <jhi-alert-error></jhi-alert-error>
+                <p jhiTranslate="artemisApp.programmingExercise.resubmitAllDialog">
+                    WARNING: Triggering all participations again is a very expensive operation. This action will start a CI build for every participation in this exercise!
```

Comment:
Atm there is no normal alert that should be displayed in the modal, as it immediately closes when confirm is pressed. I left the alert-error inside but it could also be removed for the current implementation.